File: Add Binary - Leetcode 67/Add Binary - Leetcode 67.js
var addBinary = function(a, b) {
    let x = BigInt("0b" + a);
    let y = BigInt("0b" + b);

    while (y !== 0n) {
        let answer = x ^ y;
        let carry = (x & y) << 1n;
        x = answer;
        y = carry;
    }

    return x.toString(2);
};

File: Add Binary - Leetcode 67/Add Binary - Leetcode 67.java
import java.math.BigInteger;

public class Solution {
    public String addBinary(String a, String b) {
        BigInteger x = new BigInteger(a, 2);
        BigInteger y = new BigInteger(b, 2);

        while (y.compareTo(BigInteger.ZERO) != 0) {
            BigInteger answer = x.xor(y);
            BigInteger carry = x.and(y).shiftLeft(1);
            x = answer;
            y = carry;
        }

        return x.toString(2);
    }
}
